tests/test_wfp/test_wfp.py

```diff
@@ -0,0 +1,1451 @@
+    """
+    # Mock data
+    mock_extracted_data = "mocked data"
+    mock_formatted_data = pd.DataFrame({"value": [1]})
+
+    # Mock logs, `extract_data` and `format_data` methods
+    with mock.patch(
+        "bblocks_data_importers.config.logger.info"
+    ) as mock_logger, mock.patch.object(
+        wfp_inflation_pre_load, "extract_data", return_value=mock_extracted_data
+    ) as mock_extract, mock.patch.object(
+        wfp_inflation_pre_load, "format_data", return_value=mock_formatted_data
+    ) as mock_format:
+
+        # Call `load_data` method
+        wfp_inflation_pre_load.load_data("Headline inflation (YoY)", ["USA"])
+
+        # Assertions
+        mock_extract.assert_called_once_with(1, 116)
+        mock_format.assert_called_once_with(
+            mock_extracted_data, "Headline inflation (YoY)", "USA"
+        )
+        assert (
+            wfp_inflation_pre_load._data["Headline inflation (YoY)"]["USA"] is not None
+        )
+        mock_logger.assert_any_call(
+            "Data imported successfully for indicator: Headline inflation (YoY)"
+        )
+
+
+def test_wfp_inflation_load_data_empty_data(wfp_inflation_pre_load):
+    """
+    Test that WFPInflation's `load_data` method handles empty formatted data by logging a warning and setting the data to None.
+    """
+    # Mock data
+    mock_extracted_data = "mocked data"
+    mock_empty_df = pd.DataFrame()
+
+    # Mock logs, `extract_data` and `format_data` methods
+    with mock.patch(
+        "bblocks_data_importers.config.logger.warning"
+    ) as mock_warning, mock.patch.object(
+        wfp_inflation_pre_load, "extract_data", return_value=mock_extracted_data
+    ) as mock_extract, mock.patch.object(
+        wfp_inflation_pre_load, "format_data", return_value=mock_empty_df
+    ) as mock_format:
+
+        # Call `load_data` method
+        wfp_inflation_pre_load.load_data("Headline inflation (YoY)", ["USA"])
+
+        # Assertions
+        mock_extract.assert_called_once_with(1, 116)
+        mock_format.assert_called_once_with(
+            mock_extracted_data, "Headline inflation (YoY)", "USA"
+        )
+        assert wfp_inflation_pre_load._data["Headline inflation (YoY)"]["USA"] is None
+        mock_warning.assert_any_call(
+            "No Headline inflation (YoY) data found for country - USA"
+        )
+
+
+def test_wfp_inflation_available_indicators_with_data(wfp_inflation_post_load):
+    """
+    Test that WFPInflation's `available_indicators` property returns a list of indicators when `_indicators` is populated.
+    """
+    # Mock `_indicators`
+    wfp_inflation_post_load._indicators = {
+        "Indicator A": 100,
+        "Indicator B": 200,
+    }
+
+    # Call the property
+    result = wfp_inflation_post_load.available_indicators
+
+    # Assertions
+    assert result == ["Indicator A", "Indicator B"]
+
+
+def test_wfp_inflation_available_indicators_empty(wfp_inflation_post_load):
```

Comment:
Not sure what this is testing. `_indicators` should always exist in the object. If it were an empty dictionary the extraction would fail